modules/data.atmosphere/R/debias_met_regression.R

```diff
@@ -0,0 +1,761 @@
+      # par(mfrow=c(1,1))
+      
+      
+      # Modeling the anomalies of the other predictors 
+      #  -- note: the downscaling & bias-correction of precip should have removed the monsoonal trend if there is no empirical basis for it
+      #     so this should be pretty straight-forward now
+      for(j in vars.all[vars.all!=v]){
+        met.train[met.train$ind==ind, "Q"] <- met.train[met.train$ind==ind,j]
+        met.src[met.src$ind==ind, "Q"] <- met.src[met.src$ind==ind,j]
+        
+        # Generating the predicted seasonal cycle for each variable
+        anom.train2 <- gam(Q ~ s(doy, k=6), data=met.train[met.train$ind==ind,]) 
+        anom.src2   <- gam(Q ~ s(doy, k=6), data=met.src[met.src$year %in% yrs.overlap & met.src$ind==ind,]) 
+        
+        met.train[met.train$ind==ind, paste0(j, ".anom")] <- resid(anom.train2)
+        met.src[met.src$ind==ind, paste0(j, ".anom")] <- met.src[met.src$ind==ind,"Q"] - predict(anom.src2, newdata=met.src[met.src$ind==ind,])
+        
+        rm(anom.train2, anom.src2)
+      }
+      
+      # CRUNCEP has a few variables that assume a constant pattern from 1901-1950; 
+      # so we don't want to use their anomaly as a predictor otherwise we will perpetuate that less than ideal situation
+      if(CRUNCEP==T & v %in% c("surface_downwelling_longwave_flux_in_air", "air_pressure", "wind_speed")) met.src$anom.raw <- 0
+      
+      # Actually Modeling the anomalies
+      #  -- If we have empirical data, we can pair the anomalies to find a way to bias-correct those
+      #  -- If one of our datasets is a GCM, the patterns observed are just what underly the climate signal and no actual
+      #     event is "real".  In this case we just want to leverage use the covariance our other met drivers to try and get
+      #     the right distribution of anomalies
+      if(pair.anoms==TRUE){ 
+        # if it's empirical we can, pair the anomalies for best estimation
+        # Note: Pull the covariates from the training data to get any uncertainty &/or try to correct covariances
+        #        -- this makes it mroe consistent with the GCM calculations
+        dat.anom <- merge(met.src  [met.src$ind==ind & met.src$year %in% yrs.overlap, c("year", "doy", "ind", "X", "anom.raw")], 
+                          met.train[met.train$ind==ind,c("year", "doy", "anom.train", "ind", vars.all[vars.all!=v], paste0(vars.all[vars.all!=v], ".anom"))])
+        
+        k=round(length(unique(met.src$year))/50,0)
+        k=max(k, 4) # we can't have less than 4 knots
+        
+        # plot(anom.train ~ anom.raw, data=dat.anom)
+        # abline(a=0, b=1, col="blue")
+        # abline(lm(anom.train ~ anom.raw, data=dat.anom), col="red", lty="dashed")
+        
+        # Modeling in the predicted value from mod.bias
+        dat.anom$pred <- predict(mod.bias, newdata=dat.anom)
+        
+        if (v %in% c("air_temperature", "air_temperature_maximum", "air_temperature_minimum")){
```

Comment:
This if statement with many switches is hard to follow. Would swtich statement be easier to read?

Reply:
I agree this is cumbersome and hard to follow, but am not familiar with switch syntax.  If someone can point me to a good example to follow, I can play around with it if I get a chance later.